backend/static/js/qleverUI.js

```diff
@@ -219,22 +219,35 @@ $(document).ready(function () {
   // 1. Call processQuery (sends query to backend + displays results).
   // 2. Add query hash to URL.
   //
-  $("#exebtn").click(async function() {
+  $("#exebtn").click(function() {
     log("Start processing", "other");
     $("#suggestionErrorBlock").parent().hide();
-    await processQuery(parseInt($("#maxSendOnFirstRequest").html()));
 
     // Add query hash to URL (we need Django for this, hence the POST request),
     // unless this is a URL with ?query=...
-    $.post("/api/share", { "content": editor.getValue() }, function (result) {
-      log("Got pretty link from backend", "other");
-      if (window.location.search.indexOf(result.queryString) == -1) {
-        const newUrl = window.location.origin
-                        + window.location.pathname.split("/")
-                                         .slice(0, 2).join("/") + "/" + result.link;
-        window.history.pushState("html:index.html", "QLever", newUrl);
+    const acquireShareLink = async () => {
+      const response = await fetch("/api/share", {
+        method: "POST",
+        body: new URLSearchParams({
+          "content": editor.getValue()
+        })
+      });
+      if (response.ok) {
+        const result = await response.json();
+        log("Got pretty link from backend", "other");
+        if (window.location.search.indexOf(result.queryString) == -1) {
+          const newUrl = window.location.origin
+                          + window.location.pathname.split("/")
+                                           .slice(0, 2).join("/") + "/" + result.link;
+          window.history.pushState("html:index.html", "QLever", newUrl);
+        }
       }
-    }, "json");
+    };
+
+    Promise.all([
+      processQuery(parseInt($("#maxSendOnFirstRequest").html())),
+      acquireShareLink()
+    ]).catch(error => log(error, 'requests'));
```

Comment:
Is this the idiomatic way to do a post request with `fetch`?

Why not just do fetch, then, then, catch?

Reply:
This `Promise.all` wrapper is there to make the 2 HTTP requests concurrently, as it was before.
Alternatively I could write something along the lines of this:
```js
try {
  const promise 1= processQuery(parseInt($("#maxSendOnFirstRequest").html()));
  const promise2 = acquireShareLink();
  await promise1;
  await promise2;
} catch (error) {
  log(error, 'requests');
}
```